File: src/Scripting/Lua/patterns/peer_to_peer.lua
local peers = {}
local function add_peer(name) peers[name] = { inbox = {} } end
local function send(from, to, msg) table.insert(peers[to].inbox, from .. ":" .. msg) end
add_peer("a"); add_peer("b")
send("a", "b", "hello")
assert(peers.b.inbox[1] == "a:hello")
return true

File: src/Scripting/PHP/patterns/peer_to_peer.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('peer_to_peer');
$inboxes = ['a'=>[], 'b'=>[]];
$send = static function(string $from, string $to, string $msg) use (&$inboxes): void { $inboxes[$to][] = "$from:$msg"; };
$send('a','b','hello'); $send('b','a','hi');
$check($inboxes['a'] === ['b:hi'] && $inboxes['b'] === ['a:hello']);
